c/authService.c

```diff
@@ -173,6 +174,86 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
   return 0;
 }
 
+int resetPassword(HttpService *service, HttpResponse *response) {
+  int returnCode = 0, reasonCode = 0;
+  HttpRequest *request = response->request;
+  
+  if (!strcmp(request->method, methodPOST)) {
+    char *inPtr = request->contentBody;
+    char *nativeBody = copyStringToNative(request->slh, inPtr, strlen(inPtr));
+    int inLen = nativeBody == NULL ? 0 : strlen(nativeBody);
```

Comment:
Can you do anything if `nativeBody` is NULL. Maybe you should return error right away and leave.